### PR TITLE
PM: Add missing tests to increase coverage

### PR DESCRIPTION
**Pull request by majunkier**
Cover missing tests for increase code coverage for power management : 
- pm_state_force, 
- pm_state_cpu_get_all, 
- pm_state_next_get.

### CODE REVIEW

**Review by tmleman — DISMISSED** [review]

File: tests/subsys/pm/power_mgmt/src/main.c [hunk 1]
```diff
@@ -444,6 +450,36 @@ ZTEST(power_management_1cpu, test_device_state_lock)
...
+	zassert_equal(state, 0, NULL);
+}
+
+ZTEST(power_management_1cpu, test_force_state)
+{
+	forced_state = PM_STATE_STANDBY;
+	bool ret = pm_state_force(0, &(struct pm_state_info) {forced_state, 0, 0});
+
+	zassert_equal(ret, true, "Error in force state");
+	set_pm = true;
```
Comment (ceolin) [inline tests/subsys/pm/power_mgmt/src/main.c:467]: This should be set by the notifier in the other test. You don't need to manipulate it. Instead, just return after you check this test in `pm_state_set`.
```
	if (testing_force_state) {
		/* if forced to given power state was called */
		zassert_equal(state, forced_state, NULL);
		return;
	}
```

File: tests/subsys/pm/power_mgmt/src/main.c [hunk 2]
```diff
@@ -444,6 +450,36 @@ ZTEST(power_management_1cpu, test_device_state_lock)
...
+	testing_force_state = true;
+	k_sleep(K_SECONDS(1U));
+	testing_force_state = false;
+	set_pm = false;
```
Comment (ceolin) [inline tests/subsys/pm/power_mgmt/src/main.c:471]: not needed based on previous comment.

File: tests/subsys/pm/device_runtime_api/src/main.c [hunk 3]
```diff
@@ -224,6 +224,7 @@ ZTEST(device_runtime_api, test_unsupported)
 {
 	const struct device *const dev = DEVICE_GET(pm_unsupported_device);
 
+	zassert_false(pm_device_runtime_put_async(dev), "");
```
Comment (ceolin) [inline tests/subsys/pm/device_runtime_api/src/main.c:227]: Please move it to after ` zassert_equal(pm_device_runtime_put(dev), 0, "");`

**Review by ceolin — CHANGES_REQUESTED** [review]
Very minor changes needed. Would mind also split this change into two commits since they are touching different tests ?

Other than that looks nice. Thanks.

File: tests/subsys/pm/power_mgmt/src/main.c (hunk 1 above)
Comment (majunkier) [reply tests/subsys/pm/power_mgmt/src/main.c:467]: Without this parameter set to true, test is failing on assert in pm_state_set : 
`DEBUG   - OUTPUT: START - test_force_state
DEBUG   - OUTPUT:     Assertion failed at WEST_TOPDIR/zephyr/tests/subsys/pm/power_mgmt/src/main.c:188: pm_state_set: (set_pm == true is false)
DEBUG   - OUTPUT: Notification to enter suspend was not sent to the App`

File: tests/subsys/pm/power_mgmt/src/main.c [hunk 4]
```diff
@@ -444,6 +452,36 @@ ZTEST(power_management_1cpu, test_device_state_lock)
...
+
+ZTEST(power_management_1cpu, test_state_next_get)
+{
+	static const struct pm_state_info state = {.state = PM_STATE_ACTIVE};
+	const struct pm_state_info *next;
+
+	next  = pm_state_next_get(0);
+	zassert_equal(next->state, state.state, NULL);
+}
+
```
Comment (ceolin) [inline tests/subsys/pm/power_mgmt/src/main.c:483]: This test is not really helpful. Would be nice to test some particular state selected by the policy. Unfortunately this test does not declare any power state.
Reviewing this pr I noticed that this API is not reflecting a forced power state, I am fixing it and will add a test to ensure that this works correctly.

Drop this test only please, the rest looks good to me.

**Comment by ceolin** [comment]
@majunkier https://github.com/zephyrproject-rtos/zephyr/pull/66409

**Review by ceolin — APPROVED** [review]

**Comment by majunkier** [comment]
@tmleman could you rechecked ? 

[tmleman](https://github.com/tmleman) previously approved these changes [on Nov 8](https://github.com/zephyrproject-rtos/zephyr/pull/64150#pullrequestreview-1720707103)

**Review by tmleman — APPROVED** [review]

**Pull request merged**